numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,102 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
 
     return view
 
+def sliding_window_view(x, shape, step=None, writeable=False):
+    """
+    Create sliding window views of the N dimensions array with the given window
+    shape. Window slides across each dimension of `x` and extract subsets of `x`
+    at any window position.
+
+    For some cases, there may be more efficient approaches.
+
+    Parameters
+    ----------
+    x : ndarray
+        Array to create sliding window views.
+    shape : sequence of int
+        The shape of the window. Must have same length as number of input array dimensions.
+    step: sequence of int, optional
+        The steps of window shifts for each dimension on input array at a time.
+        If given, must have same length as number of input array dimensions.
+        Defaults to 1 on all dimensions.
+    writeable : bool, optional
```

Comment:
We called it readonly in the broadcast_to internal function, so maybe that is the better choice even?

Reply:
There was some discussion back then on the choice, do not remember though.